app/src/org/commcare/android/database/app/models/FormDefRecord.java

```diff
@@ -0,0 +1,287 @@
+package org.commcare.android.database.app.models;
+
+import android.database.Cursor;
+import android.database.SQLException;
+import android.os.Environment;
+
+import org.apache.commons.lang3.StringUtils;
+import org.commcare.CommCareApplication;
+import org.commcare.android.storage.framework.Persisted;
+import org.commcare.models.database.SqlStorage;
+import org.commcare.models.framework.Persisting;
+import org.commcare.modern.database.Table;
+import org.commcare.modern.models.MetaField;
+import org.commcare.provider.FormsProviderAPI;
+import org.commcare.utils.FileUtil;
+
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Vector;
+
+@Table(FormDefRecord.STORAGE_KEY)
+public class FormDefRecord extends Persisted {
+
+    public static final String STORAGE_KEY = "form_def";
+
+    public static final String META_DISPLAY_NAME = "displayName";
```

Comment:
Did we audit whether we need to retain all of these? I think many/most of these properties can actually be eliminated, but i could be wrong about that

Reply:
Yeah, Some of these are actually not getting used right now. I think I was being overly cautious to not create unnecessary changes as part of this migration. Can remove these now though.

Reply:
@ctsims Here is a list of all fields that are not getting used currently - 

Instance Provider/ Instance Record

* submissionUri
* date (last status change date)
* displaySubtext

Form Def 

* description
* submissionUri
* displaySubtext
* md5Hash
* date
* jrcacheFilePath
* language

Just want to make sure we are fine with removing all of these.

Reply:
Reg: "last status change date": You are right. We had 2 date fields one in FormRecord and other in InstacneRecord. Now in the merged FormRecord I have retained the original FormRecord field which also gets used for form ordering. 

Reg: "private key" : There is no private key currently in FormDefRecord. So guessing we already removed it earlier. Though we do use the public key [here](https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/utils/EncryptionUtils.java#L131)

Reply:
I am 99% that EncryptedFormInformation stuff is only used by ODK. That class is only called while reading forms currently (and we've never written any that way), if you trace the code back, so I think we can remove it without concern.

Reply:
Cool. Going to remove the `EncryptionUtils.getEncryptedFormInformation` as well since that will always return null in case of no publick key